dnstable_convert.c

```diff
@@ -108,29 +112,120 @@ static uint64_t			count_entries_merged;
 	case WDNS_TYPE_TA:                 \
 	case WDNS_TYPE_DLV:
 
+/* Display current soft- and hard-limits for # descriptors. */
+static int
+show_fd_limit(FILE *fp, struct rlimit *rl)
+{
+	int ret = getrlimit(RLIMIT_NOFILE, rl);
+
+	if (ret == 0)
+		fprintf(fp, "fd-limit: soft=%lu, hard=%lu\n",
+			(unsigned long) rl->rlim_cur, (unsigned long) rl->rlim_max);
+	else
+		fprintf(fp, "getrlimit() failed, error=%d\n", errno);
+
+	return(ret);
+}
+
+/* Check/increment the descriptor resource limit. */
+static void
+check_fd_limit(FILE *fp)
+{
+	struct rlimit rl;
+	int ret;
+
+	fputs("Checking resource limits\n", fp);
+
+	ret = show_fd_limit(fp, &rl);
+
+	if (ret == 0) {
+		/* Increase soft-limit to hard-limit. */
+		if (rl.rlim_cur < rl.rlim_max) {
+			rl.rlim_cur = rl.rlim_max;
+
+			ret = setrlimit(RLIMIT_NOFILE, &rl);
+			if (ret == 0) {
+				fprintf(fp, "fd-limit: Updated soft-limit\n");
+				show_fd_limit(fp, &rl);
+			}
+			else
+				fprintf(fp, "setrlimit() failed, error=%d\n", errno);
+		}
+	}
+}
+
+static void
+abrt_handler(int sig __attribute__((unused)))
+{
+	int err = errno;
+
+	fprintf(stderr, "ABRT handler called, errno=%d (%s)\n", err, strerror(err));
+	/* Return, allowing a core-file to be produced. */
+}
+
+/* Catch certain signals. */
+static void
+setup_handlers(void)
+{
+	struct sigaction sa = { 0 };
+
+	sigemptyset(&sa.sa_mask);
+
+	/* Catch any assert() failures generated in other libraries. */
+	sa.sa_handler = abrt_handler;
+	sigaction(SIGABRT, &sa, NULL);
+}
+
+/* Show details about startup environment. */
+static int
+show_startup_details(FILE *fp)
+{
+	struct utsname un;
+	struct timeval tv;
+	char timebuf[64];
+	char *startdir;
+
+	uname(&un);
+	gettimeofday(&tv, NULL);
+
+	ctime_r(&tv.tv_sec, timebuf);
+
+	fprintf(fp, "Start: %s", timebuf);
+	fprintf(fp, "Host: %s\n", un.nodename);
+	fprintf(fp, "Kernel: %s %s %s %s\n", un.sysname, un.machine, un.release, un.version);
+	fprintf(fp, "Run by: UID=%u EUID=%u\n", getuid(), geteuid());
+	startdir = get_current_dir_name();
+	fprintf(fp, "Start dir: %s\n", startdir);
+	free(startdir);
+	fputc('\n', fp);
+}
+
 static void
 do_stats(void)
 {
 	struct timespec dur;
 	double t_dur;
+	int fd;
 
 	nmsg_timespec_get(&dur);
 	nmsg_timespec_sub(&start_time, &dur);
 	t_dur = nmsg_timespec_to_double(&dur);
+	fd = dup(2);
```

Comment:
Is it leftover debug code to print the next file descriptor value in this function?

Reply:
Nope, this is new debug code (to provide better logging when nearing or hitting fd limits)